jbmc/src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1614,13 +1614,14 @@ exprt java_bytecode_parsert::get_relement_value()
 
   case '[':
     {
-      array_exprt values;
       u2 num_values=read_u2();
+      exprt::operandst values;
+      values.reserve(num_values);
       for(std::size_t i=0; i<num_values; i++)
       {
-        values.operands().push_back(get_relement_value());
+        values.push_back(get_relement_value());
       }
-      return std::move(values);
+      return array_exprt(std::move(values), array_typet(typet(), exprt()));
```

Comment:
Not sure about this one -- somehow an empty array with a correct type (seen many times below) seems considerably safer than this partially-constructed thing. If we're going to partially construct something, we should probably use a constructor with that express purpose, rather than pretending we're doing it safely?

Reply:
The language front-ends will always remain a bit weird. I'm not sure what the best option is - we could also do `exprt values(ID_array);` to make very clear we are not building anything like a complete expression here.